data/cluster_id.go

```diff
@@ -50,7 +51,7 @@ func NewClusterIDFromPersistentStorage(sgc KubeSecretGetterCreator) ClusterID {
 // Get is the ClusterID interface implementation
 func (c clusterIDFromPersistentStorage) Get() (string, error) {
 	secret, err := c.secretGetterCreator.Get(wfmSecretName)
-	if err != nil {
+	if err != nil && !apierrors.IsNotFound(err) {
```

Comment:
can you add a comment here explaining why this is here? I believe I accidentally removed the equivalent logic a few weeks ago. also, if there's no way to add appropriate tests for this, can you add an issue to add them later?